packages/core/src/lib/template/template.interface.ts

```diff
@@ -6,11 +6,16 @@ export interface ITemplate {
   messages: IMessage[];
 }
 
+export interface IMessageValidator {
+  validate(payload: ITriggerPayload): Promise<boolean> | boolean;
+}
+
 export interface IMessage {
   subject?: string;
   channel: ChannelTypeEnum;
   template: string | ((payload: ITriggerPayload) => Promise<string> | string);
   active?: boolean | ((payload: ITriggerPayload) => Promise<boolean> | boolean);
+  validator?: IMessageValidator;
```

Comment:
@davidsoderberg @ComBarnea message level validation is a good idea, I wonder if there is a use-case for template-wide validator?

Reply:
How would the prio be between message level and template level validator?

Reply:
Cascading, message level will win. But I'm not sure the template level is even needed. We can start with a message level and later expand if we see a demand.